### PR TITLE
Add to `ujust harden-flatpak` logic that applies the highest supported hwcap

### PR DESCRIPTION
**Pull request by spaceoden**
This script runs ld-linux-x86-64.so.2 --help to identify supported microarchitectures, then applies a flatpak override to set LD_PRELOAD to load the highest supported hwcap for libhardened_malloc.so

### CODE REVIEW

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 1]
```diff
@@ -177,3 +177,13 @@ toggle-anticheat-support:
...
         echo "The sysctl hardening file is missing the ptrace_scope setting."
     fi
+
+
+# Detect microarchitecture support and apply highest supported flatpak malloc hwcap
+set-flatpak-malloc-hwcap:
+    #!/usr/bin/pkexec /usr/bin/bash
+    uarches=$(/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help | grep '(supported, searched)' | cut -d'v' -f2)
+    bestuarch=${uarches:0:1}
+    if [ ! -z "${bestuarch}" ]
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just]: nitpick: `if ; then`

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 2]
```diff
@@ -177,3 +177,13 @@ toggle-anticheat-support:
     else 
         echo "The sysctl hardening file is missing the ptrace_scope setting."
     fi
+
+
+# Detect microarchitecture support and apply highest supported flatpak malloc hwcap
+set-flatpak-malloc-hwcap:
+    #!/usr/bin/pkexec /usr/bin/bash
+    uarches=$(/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help | grep '(supported, searched)' | cut -d'v' -f2)
+    bestuarch=${uarches:0:1}
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just]: can you either add a comment explaining this or make it self-documenting?

File: config/files/usr/share/ublue-os/just/60-custom.just [hunk 3]
```diff
@@ -177,3 +177,13 @@ toggle-anticheat-support:
     else 
         echo "The sysctl hardening file is missing the ptrace_scope setting."
     fi
+
+
+# Detect microarchitecture support and apply highest supported flatpak malloc hwcap
+set-flatpak-malloc-hwcap:
```
Comment (RoyalOughtness) [inline config/files/usr/share/ublue-os/just/60-custom.just]: maybe `optimize-flatpak-hardened-malloc`?

**Comment by RoyalOughtness** [comment]
Thanks for this! I left some comments

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 2 above)
Comment (spaceoden) [reply config/files/usr/share/ublue-os/just/60-custom.just]: `/var/run/host/usr/lib64/ld-linux-x86-64.so.2 --help` prints help info for the ld.so linker. Helpfully, it also reports detected glibc-hwcaps. I `grep` those, then use `cut` and the substring selection (`:0:1`) to get an integer 2-4 corresponding to the system's highest supported hwcap. On an ancient x86_64 system with no v2 support, the `bestuarch` variable is left empty. The `if` statement checks if it is empty, and if not, it updates the LD_PRELOAD override to the hwcap with our 2-4 integer.

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 3 above)
Comment (spaceoden) [reply config/files/usr/share/ublue-os/just/60-custom.just]: done

File: config/files/usr/share/ublue-os/just/60-custom.just (hunk 1 above)
Comment (spaceoden) [reply config/files/usr/share/ublue-os/just/60-custom.just]: done

**Comment by spaceoden** [comment]
I just realized there's a `harden-flatpak` ujust command. I don't think my hwcap selection logic should be separate from it. If the two separate commands are present, a user may inadvertently remove optimization by running `ujust harden-flatpak` after `ujust optimize-flatpak-hardened-malloc`, replacing the LD_PRELOAD envar with the default (v1) value.

I think it's best we instead add my new optimization logic to the existing `harden-flatpak` ujust command.

**Comment by RoyalOughtness** [comment]
> I just realized there's a `harden-flatpak` ujust command. I don't think my hwcap selection logic should be separate from it. If the two separate commands are present, a user may inadvertently remove optimization by running `ujust harden-flatpak` after `ujust optimize-flatpak-hardened-malloc`, replacing the LD_PRELOAD envar with the default (v1) value.
> 
> I think it's best we instead add my new optimization logic to the existing `harden-flatpak` ujust command.

agreed

**Comment by RoyalOughtness** [comment]
@spaceoden please tag me once this is ready for review

**Comment by spaceoden** [comment]
@qoijjj its ready for review

**Comment by RoyalOughtness** [comment]
@spaceoden you still have merge conflicts

**Comment by spaceoden** [comment]
made a new PR #346

**Pull request closed without merging**